docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,297 @@
+```
+
+### Parameters
+
+* **named_expression**
+
+    An expression with an assigned name. In general, it denotes a column expression.
+
+    **Syntax:** `expression [AS] [alias]`
+
+* **row_format**    
+
+    Use the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **SERDEPROPERTIES**
+
+    A list of key-value pairs that is used to tag the SerDe definition.
+
+* **DELIMITED**
+
+    The `DELIMITED` clause can be used to specify the native SerDe and state the delimiter, escape character, null character and so on.
+    
+* **FIELDS TERMINATED BY**
+
+    Used to define a column separator.
+    
+* **COLLECTION ITEMS TERMINATED BY**
+
+    Used to define a collection item separator.
+   
+* **MAP KEYS TERMINATED BY**
+
+    Used to define a map key separator.
+    
+* **LINES TERMINATED BY**
+
+    Used to define a row separator.
+    
+* **NULL DEFINED AS**
+
+    Used to define the specific value for NULL.
+    
+* **ESCAPED BY**
+
+    Used for escape mechanism.
+
+* **RECORDREADER**
+
+    Specifies a custom RecordReader for one table.
+
+* **RECORDWRITER**
+
+    Specifies a custom RecordWriter for one table.
+
+* **recordReader_class**
+
+    Specifies a fully-qualified class name of a custom RecordReader. 
+    Default value is `org.apache.hadoop.hive.ql.exec.TextRecordReader`
+
+* **recordWriter_class**
+
+    Specifies a fully-qualified class name of a custom RecordWriter. 
+    Default value is `org.apache.hadoop.hive.ql.exec.TextRecordWriter`.
+
+* **script**
+
+    Specify a command to process data.
+
+* **boolean_expression**
+
+    Specifies any expression that evaluates to a result type `boolean`. Two or
+    more expressions may be combined together using the logical
+    operators ( `AND`, `OR` ).
+
+* **expression**
+
+    Specifies combination of one or more values, operators and SQL functions that results in a value.
+
+### Without Hive support Mode
+
+Now Spark Script transform can run without `-Phive` or `SparkSession.builder.enableHiveSupport()`.
+In this case, now we only use script transform with `ROW FORMAT DELIMIT` and treat all value passed
+to script as string. 
+
+### With Hive Support Mode
+
+When build Spark with `-Phive` and start Spark SQL with `enableHiveSupport()`, we can use script 
+transform with Hive SerDe and both `ROW FORMAT DELIMIT`.
+
+### Schema-less Script Transforms
+
+If there don't have AS clause after USING my_script, Spark assumes that the output of the script contains 2 parts:
+
+   1. key: which is before the first tab, 
+   2. value: which is the rest after the first tab.
+
+Note that this is different from specifying AS key, value because in that case, value will only contain the portion
```

Comment:
AS -> an AS
"key, value"
Maybe tick-quote all these reserved words too, like NULL as well?
value -> the value

Reply:
> AS -> an AS
> "key, value"
> Maybe tick-quote all these reserved words too, like NULL as well?
> value -> the value

Hmm use back-quota may be better?